pandas/tests/extension/base/interface.py

```diff
@@ -67,3 +67,7 @@ def test_no_values_attribute(self, data):
         # code, disallowing this for now until solved
         assert not hasattr(data, 'values')
         assert not hasattr(data, '_values')
+
+    def test_is_numeric_honored(self, data):
+        result = pd.Series(data)
+        assert result._data.blocks[0].is_numeric is data.dtype._is_numeric
```

Comment:
Maybe you can also test `df._get_numeric_data` ?

It would be nice if this would also work for `df.select_dtypes`, but I suppose that is yet another issue

Reply:
> Maybe you can also test df._get_numeric_data ?

ah, that is done below